File: JPA/hello-spring-data-jpa/src/main/java/me/kzv/datajpa/entity/Team.java
package me.kzv.datajpa.entity;

import lombok.*;

import javax.persistence.*;
import java.util.ArrayList;
import java.util.List;

@Entity
@Setter @Getter @ToString(exclude = "members")
@NoArgsConstructor(access = AccessLevel.PROTECTED)
@AllArgsConstructor
public class Team {

    @Id @GeneratedValue
    @Column(name = "team_id")
    private Long id;
    private String name;

    @OneToMany(mappedBy = "team")
    private List<Member> members = new ArrayList<>();

    public Team(String name) {
        this.name = name;
    }
}

File: JPA/kotlin-jpa-querydsl/src/main/kotlin/me/kzv/kotlinjpaquerydsl/entity/Team.kt
package me.kzv.kotlinjpaquerydsl.entity

import jakarta.persistence.Entity
import jakarta.persistence.GeneratedValue
import jakarta.persistence.Id
import jakarta.persistence.OneToMany

@Entity
class Team(
    @Id @GeneratedValue
    val id: Long? = null,
    var name: String,

    @OneToMany(mappedBy = "team") // 주인이 아니므로 업데이트 하지 않음
    val members: MutableList<Member> = mutableListOf()
){

    override fun toString(): String {
        return "Team(id=$id, name='$name')"
    }
}
